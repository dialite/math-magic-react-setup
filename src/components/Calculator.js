/* eslint-disable operator-linebreak */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prefer-stateless-function */
import React, { useState } from 'react';
import '../App.css';
import calculate from '../logic/calculate';
import '../logic/operate';

const Calculator = () => {
  const [data, setData] = useState({
    total: 0,
    next: null,
    operation: null,
    err: null,
  });

  const clickHandler = (e) => {
    setData((prevData) => calculate(prevData, e.target.name));
  };

  return (
    <div className="main-container">
      <h2>Lets Do Some Math</h2>
      <div className="calculation-field">
        <h1 className="answer" data-testid="testH2">
          {data.next || data.operation || data.total || data.err || 0}
        </h1>
        <div className="numbers-and-symbols">
          <button
            type="button"
            onClick={clickHandler}
            className="AC size"
            name="AC"
          >
            AC
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="plus-or-minus size"
            name="+/-"
          >
            +/-
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="percentage size"
            name="%"
          >
            %
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="division size orabg-bg"
            name="รท"
          >
            รท
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="seven size"
            name="7"
          >
            7
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="eight size"
            name="8"
          >
            8
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="nine size"
            name="9"
          >
            9
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="cross size orabg-bg"
            name="x"
          >
            x
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="four size"
            name="4"
          >
            4
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="five size"
            name="5"
          >
            5
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="six size"
            name="6"
          >
            6
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="minus size orabg-bg"
            name="-"
          >
            -
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="one size"
            name="1"
          >
            1
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="two size"
            name="2"
          >
            2
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="three size"
            name="3"
          >
            3
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="plus size orabg-bg"
            name="+"
          >
            +
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="zero"
            name="0"
          >
            0
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="dot size"
            name="."
          >
            .
          </button>
          <button
            type="button"
            onClick={clickHandler}
            className="equal size orabg-bg"
            name="="
          >
            =
          </button>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
