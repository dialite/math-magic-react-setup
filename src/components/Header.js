/* eslint-disable react/void-dom-elements-no-children */
import React from 'react';
import { Link } from 'react-router-dom';
import '../App.css';
import style from './styles.module.css';

const Header = () => (
  <div className="Header">
    <h1>Math Magician</h1>
    <ul className="navigation-container">
      <li>
        <Link className={style.linkStyle} to="/">
          Home
        </Link>
      </li>
      <li>
        <Link className={style.linkStyle} to="/Calculator">
          Calculator
        </Link>
      </li>
      <li>
        <Link className={style.linkStyle} to="/Quote">
          Quote
        </Link>
      </li>
    </ul>
  </div>
);

export default Header;
