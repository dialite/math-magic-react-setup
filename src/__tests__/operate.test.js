import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import operate from '../logic/operate';
import Header from '../components/Header';
import Calculator from '../components/Calculator';

test('Render NavBar', () => {
  const renderHeader = renderer
    .create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    )
    .toJSON();
  expect(renderHeader).toMatchSnapshot();
});

test('Render Calculator', () => {
  const renderCalculator = renderer.create(<Calculator />).toJSON();
  expect(renderCalculator).toMatchSnapshot();
});

test('Sum 1, 2 result will be 2', () => {
  const result = operate(1, 2, 'x');
  expect(result).toBe('2');
});

test("Should return 'can not divide by 0 message'", () => {
  const result = operate(2, 0, 'รท');
  expect(result).toBe("Can't divide by 0.");
});

test('Should display an invalid operation message when operation out of the calculator operation is used', () => {
  /* const result = operate(67, 90, '/'); */
  expect(() => operate(67, 90, '/')).toThrow("Unknown operation '/'");
});
