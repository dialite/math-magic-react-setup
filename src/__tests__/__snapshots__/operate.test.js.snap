// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Calculator 1`] = `
<div
  className="main-container"
>
  <h1
    className="title-cal"
  >
    Let s do some math!
  </h1>
  <div
    className="container"
  >
    <div
      className="input"
      data-testid="testH2"
    >
      0
    </div>
    <div
      className="keypad"
    >
      <button
        name="A/C"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        id="divide"
        name="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        id="mult"
        name="*"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        id="less"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        id="plus"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        id="equal"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
      <button
        id="cero"
        name="0"
        onClick={[Function]}
        type="button"
        value="0"
      >
        0
      </button>
    </div>
  </div>
</div>
`;

exports[`Render NavBar 1`] = `
<header
  className="header"
>
  <h1
    className="logo"
  >
    Math Magicians
  </h1>
  <div>
    <ul
      className="ul-nav"
    >
      <li
        className="link-item"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="link-item"
      >
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="link-item"
      >
        <a
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
  <footer
    className="footer"
  >
    Copyright...
  </footer>
</header>
`;
